### PR TITLE
Refactor Robot::setFromIK + multi-target IK

### PR DESCRIPTION
**Pull request by zkingston**

### CODE REVIEW

File: robowflex_library/src/robot.cpp [hunk 1]
```diff
@@ -579,84 +556,150 @@ bool Robot::hasJoint(const std::string &joint) const
...
-        [&scene, &verbose](robot_state::RobotState *state,            //
-                           const moveit::core::JointModelGroup *jmg,  //
-                           const double *values)                      //
+    bool sampled = true;
+    for (std::size_t j = 0; j < n and sampled; ++j)
     {
-        state->setJointGroupPositions(jmg, values);
-        state->updateCollisionBodyTransforms();
+        const auto &point = regions[j]->sample();
+        if ((sampled &= point.first))
```
Comment (ChamzasKonstantinos) [inline robowflex_library/src/robot.cpp:651]: Is double parenthesis needed here?

**Review by ChamzasKonstantinos — APPROVED** [review]
Just some minor comments. Looks good otherwise. This really expands  the Ik request repertoire

File: robowflex_library/src/robot.cpp (hunk 1 above)
Comment (zkingston) [reply robowflex_library/src/robot.cpp:651]: yeah, the double is needed so it actually sets `sampled`

File: robowflex_library/include/robowflex_library/robot.h [hunk 2]
```diff
@@ -340,74 +340,184 @@ namespace robowflex
...
-         *  \param[in] tolerances Tolerance about \a orientation.
-         *  \return True on success, false on failure.
+        /** \brief Get the tip frames for the IK solver for a given joint model group \a group.
+         *  \param[in] group The group to get the tip frames for.
+         *  \return The list of tip frames. Will return an empty list on error.
          */
-        bool setFromIK(const std::string &group, const GeometryConstPtr &region, const RobotPose &pose,
-                       const Eigen::Quaterniond &orientation,
-                       const Eigen::Vector3d &tolerances = constants::ik_rot_tolerance);
+        std::vector<std::string> getSolverTipFrames(const std::string &group) const;
```
[the remark this answers is not in the document]
Comment (zkingston) [reply robowflex_library/include/robowflex_library/robot.h:514]: End-effectors are a separate concept in MoveIt (which are defined in the SRDF) so I'd rather keep it precise.

**Pull request merged**